### ISSUE TITLE
Trying to use cache with bundle install inside docker container

### ISSUE THREAD

**Issue by texdade** [rubygems/rubygems#7902]
### Describe the problem as clearly as you can
Hello, I'm trying to reduce the time it takes to run `bundle install` in my docker container. The layer currently takes ~5 mins every time the Gemfile is modified. I'm trying to mount a cache volume using `--mount=type=cache,target=/opt/app/vendor` and running `bundle package && bundle install -local`.  The whole command looks like this:

```
RUN --mount=type=cache,target=/opt/app/vendor bundle config without "development test" \
  && bundle package \
  && bundle install --local
```

However, this does not work, when I modify the Gemfile (e.g. adding a Gem), this still repeats the whole gem installation (or at least the `bundle package` step). Is there a way to tell bundler to use the saved cache without re-doing the whole job each time? Currently the whole thing seems a bit counterintuitive.

### Did you try upgrading rubygems & bundler?
Bundler version is `Bundler version 2.3.11`

### Which command did you run?

From my `dockerfile`:
```
RUN --mount=type=cache,target=/opt/app/vendor bundle config without "development test" \
  && bundle package \
  && bundle install --local
```

### What were you expecting to happen?

I expected bundler to use the cache when installing new gem, and to not reinstall all the new gems from scratch

### What actually happened?

Bundler takes a long time, reinstalling all the gems

**Comment by deivid-rodriguez** [comment]
Can you start by upgrading to the latest Bundler. After that, can you share an actual Dockerfile that reproduces the problem? Also, can you share the output of `bundle package` and how long it takes? It is expected that if you change the `Gemfile`, `bundle package` will re-resolve gems, update the lockfile, and then update your application gem cache, removing no longer necessary `.gem` files, and adding new ones.

**Comment by deivid-rodriguez** [comment]
Closing due to lack of feedback.

**Comment by texdade** [comment]
I'm very sorry but I lost the notification of your messages and never answered.
Apparently I had two separate issues: one was technical and one was not. The first was that I had to mount the cache in `/app/vendor` and not `/opt/app/vendor`. This is why I could see gems getting fetched even if nothing had changed.
The second issue was an assumption I made: I thought that, when using bundle package, gems wouldn't be installed. I mistakenly thought that the gem cache would save me installation time, but what it does instead is skip the download of the gem.

Anyhow, all good. Thanks again ^^
